### PR TITLE
Add -qnoopt -qnosmp flags to any OLCF machine (Ascent)

### PR DESCRIPTION
**Pull request by amametjanov**
Adding the necessary compiler flags to build pioc_support.c with the
IBM XL compiler on OLCF machines (Ascent)

Addresses E3SM-Project/scorpio#215, E3SM-Project/scorpio#220

### CODE REVIEW

**Comment by amametjanov** [comment]
This is needed to avoid ICE build issue in `e3sm_integration` tests on OLCF Ascent (open-access clone of Summit). 
Tested on `SMS.f19_g16.X.ascent_ibm` with [2021-May-03 e3sm master](https://github.com/E3SM-Project/E3SM/commit/9c9f3d532e495c274f10e6bcd1d7c06277a58ab5) with submodule pointing to [2021-Mar-12 scorpio](https://github.com/E3SM-Project/scorpio/tree/7709c066b09068a15d5dcded92af16be495bed51).

File: src/clib/CMakeLists.txt [hunk 1]
```diff
@@ -219,7 +219,7 @@ endif ()
 # Hack to circumvent IBM XL (16.1.1-3) internal compiler error by
 # disabling optimization for pioc_support.c
 cmake_host_system_information (RESULT FQDN_SITENAME QUERY FQDN)
-if (FQDN_SITENAME MATCHES "^.*summit[.]olcf")
+if (FQDN_SITENAME MATCHES "^.*[.]olcf")
```
Comment (jayeshkrishna) [inline src/clib/CMakeLists.txt:222]: Do you think we should add these flags for just ascent and summit or all olcf machines (for XL)?

Comment (amametjanov) [reply src/clib/CMakeLists.txt:222]: I think it should be all: (1) it wasn't easy to find this (the flags do not show up in config_compilers.xml and Depends.* files and GH search does not recursively search in submodules), and (2) any XL version from 16.1.1-1 to 16.1.1-10 errors out with ICE.

Comment (jayeshkrishna) [reply src/clib/CMakeLists.txt:222]: ok, sounds good.

**Review by jayeshkrishna — APPROVED** [review]

**Pull request merged**